airflow/configuration.py

```diff
@@ -121,9 +121,29 @@ class AirflowConfigParser(ConfigParser):
         ('core', 'sql_alchemy_conn'),
         ('core', 'fernet_key'),
         ('celery', 'broker_url'),
-        ('celery', 'result_backend')
+        ('celery', 'result_backend'),
+        # Todo: remove this in Airflow 1.11
+        ('celery', 'celery_result_backend'),
     }
 
+    # A two-level mapping of (section -> new_name -> old_name). When reading
+    # new_name, the old_name will be checked to see if it exists. If it does a
+    # DeprecationWarning will be issued and the old name will be used instead
+    deprecated_options = {
```

Comment:
How are multi-hop upgrades handled?
Say Airflow 1.8 -> 1.11

Reply:
@afernandez Traditionally, we have not supported them very well. We have a few pre apache releases that required pretty much a full reinstall. Right now, I think we should mostly offer to support one hop upgrades.

Reply:
Yeah - right now one hop is the most we should aim for. One of my long-term tasks is to write down what the back-compat policy we should adopt is. (I plan on basing it on Django's when I get around to it)

Reply:
Breaking changes should happen only in major versions. Thus not from 1.10 -> 1.11, but rather from 1.10 -> 2.0